Beta docs (#900)
microsoftgraph/entra-powershell: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaCustomSecurityAttributeDefinition.md

# New-EntraBetaCustomSecurityAttributeDefinition

## Synopsis

Create a new customSecurityAttributeDefinition object.

## Syntax

```powershell
New-EntraBetaCustomSecurityAttributeDefinition 
 -IsSearchable <Boolean> 
 -IsCollection <Boolean> 
 -AttributeSet <String> 
 -Type <String> 
 -Name <String> 
 -Status <String>
 -UsePreDefinedValuesOnly <Boolean> 
 [-Description <String>]
 [<CommonParameters>]
```

## Description

> Create a new customSecurityAttributeDefinition object. Specify `IsSearchable`, `IsCollection`, `AttributeSet`, `Type`, `Name`, `Status` and `UsePreDefinedValuesOnly` parameters for create a new custom security attribute definition.

Reviewer: Changes done

## Examples

### Example 1: Add a custom security attribute

```powershell
Connect-Entra -Scopes 'CustomSecAttributeDefinition.Read.All, CustomSecAttributeDefinition.ReadWrite.All'

> $params = @{
>     Name = 'ProjectTest' 
>     Description = 'Target completion' 
>     Type = 'String' 
>     Status = 'Available' 
>     AttributeSet = 'Test' 
>     IsCollection = $False 
>     IsSearchable = $True 
>     UsePreDefinedValuesOnly = $False
> }
> New-EntraBetaCustomSecurityAttributeDefinition  @params

Reviewer 2: ```suggestion
$AttributeSet  = Get-EntraBetaAttributeSet -Id '<CustomAttributeSetId>'
$params = @{
    Name = 'ProjectTest' 
    Description = 'Target completion' 
    Type = 'String' 
    Status = 'Available' 
    AttributeSet = $AttributeSet.Id 
    IsCollection = $False 
    IsSearchable = $True 
    UsePreDefinedValuesOnly = $False
}
New-EntraBetaCustomSecurityAttributeDefinition  @params
```

Reviewer: changes done

```

```Output
Id               AttributeSet Description       IsCollection IsSearchable Name        Status    Type   UsePreDefinedValuesOnly
--               ------------ -----------       ------------ ------------ ----        ------    ----   -----------------------
Test_ProjectTest Test         Target completion False        True         ProjectTest Available String False
```

This example demonstrates how to add a custom security attribute.

- `Name` parameter specifies the name of the custom security attribute.

> - `Description` parameter specifies the description for the attribute set.

Reviewer 2: ```suggestion
- `Description` parameter specifies the description of the custom security attribute.
```

Reviewer: changes done

 -Type <String> 
 -Name <String> 
 -Status <String>
 -UsePreDefinedValuesOnly <Boolean> 
 [-Description <String>]
 [<CommonParameters>]
```

## Description

Create a new customSecurityAttributeDefinition object. Specify `IsSearchable`, `IsCollection`, `AttributeSet`, `Type`, `Name`, `Status` and `UsePreDefinedValuesOnly` parameters for create a new custom security attribute definition.

You can define up to 500 active objects in a tenant.

In delegated scenarios with work or school accounts, the signed-in user must be assigned a supported Microsoft Entra role or a custom role with the necessary permissions. The Attribute Definition Administrator is the only privileged role supported for this operation.

## Examples

### Example 1: Add a custom security attribute

```powershell
Connect-Entra -Scopes 'CustomSecAttributeDefinition.Read.All', 'CustomSecAttributeDefinition.ReadWrite.All'
$AttributeSet  = Get-EntraBetaAttributeSet -Id '<CustomAttributeSetId>'
$params = @{
    Name = 'ProjectTest' 
    Description = 'Target completion' 
    Type = 'String' 
    Status = 'Available' 
    AttributeSet = $AttributeSet.Id 
    IsCollection = $False 
    IsSearchable = $True 

>     UsePreDefinedValuesOnly = $False

Reviewer 2: ```suggestion
    UsePreDefinedValuesOnly = $True
```

```

### -Description

> Description of the custom security attribute. Can be up to 128 characters long and include Unicode characters. Can be changed later.

Reviewer 2: ```suggestion
Description of the custom security attribute, up to 128 characters long and including Unicode characters. This description can be changed later.
```

Reviewer: changes done
